Support modem discovery via automatic modem implementation (#143)
nerves-networking/vintage_net_mobile: README.md

  }
```

> ## Automatic Modem

Reviewer: ```suggestion
## Automatic modem detection
```

> The `VintageNetMobile.Modem.Automatic` implementation is a modem implementation
> that will try to discover the attached modem. This is useful if you need to
> support a range of modems and cannot provide a static configuration for one
> particular modem.

Reviewer: ```suggestion
If it is not known ahead of time what modem will be installed, the `VintageNetMobile.Modem.Automatic` implementation can detect and configure it for you. It works by waiting for the system to report a USB device
with a known vendor and product ID. Once one is found, it calls `VintageNet.configure/3` to change the configuration from using the `Automatic` modem to the actual modem implementation. It tells `VintageNet.configure/3` to not persist the configuration so that the detection process runs again on the next boot. 

Here's an example:
```

```elixir
  %{
    type: VintageNetMobile,
    vintage_net_mobile: %{

>       modem: VintageNetMobile.Modme.Automatic,

Reviewer: ```suggestion
      modem: VintageNetMobile.Modem.Automatic,
```

      service_providers: [
        %{apn: "wireless.twilio.com"}
      ]
    }
  }
```

> Currently supported autodetecting modems are:
> 
> 1. `Quectel EC21`
> 1. `Quectel EC25`
> 1. `Quectel BG96`
> 1. `Telit LE910`

Reviewer: ```suggestion
See [VintageNetMobile.Modem.Automatic.Discovery](source link) for list of modems that can be detected.
```
